evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -382,3 +390,54 @@ def _decompose_target(X, y, fh):
             for colname in y.columns:
                 result_dfs.append(_decompose_target(X, y[colname], None))
         return result_dfs
+
+    def get_trend_prediction_intervals(self, y, coverage=None):
+        """Calculate the prediction intervals for the trend data.
+
+        Args:
+            y (pd.Series): Target data.
+            coverage (list[float]): A list of floats between the values 0 and 1 that the upper and lower bounds of the
+                prediction interval should be calculated for.
+
+        Returns:
+            dict of pd.Series: dict: Prediction intervals, keys are in the format {coverage}_lower or {coverage}_upper.
+        """
+        if coverage is None:
+            coverage = [0.95]
+
+        self._check_oos_past(y)
+        alphas = [1 - val for val in coverage]
+
+        if not self.forecast_summary or len(y) != len(
+            self.forecast_summary.predicted_mean,
+        ):
+            self._project_trend_and_seasonality(y)
+
+        prediction_interval_result = {}
+        for i, alpha in enumerate(alphas):
+            result = self.forecast_summary.summary_frame(alpha=alpha)
+            overlapping_ind = [ind for ind in y.index if ind in result.index]
+            prediction_interval_result[f"{coverage[i]}_lower"] = (
+                result["mean_ci_lower"] - result["mean"]
+            )
+            prediction_interval_result[f"{coverage[i]}_upper"] = (
+                result["mean_ci_upper"] - result["mean"]
+            )
+            if len(overlapping_ind) > 0:  # y.index is datetime
+                prediction_interval_result[
+                    f"{coverage[i]}_lower"
+                ] = prediction_interval_result[f"{coverage[i]}_lower"][overlapping_ind]
+                prediction_interval_result[
+                    f"{coverage[i]}_upper"
+                ] = prediction_interval_result[f"{coverage[i]}_upper"][overlapping_ind]
+            else:  # y.index is not datetime (e.g. int)
+                prediction_interval_result[
+                    f"{coverage[i]}_lower"
+                ] = prediction_interval_result[f"{coverage[i]}_lower"][-len(y) :]
+                prediction_interval_result[
+                    f"{coverage[i]}_upper"
+                ] = prediction_interval_result[f"{coverage[i]}_upper"][-len(y) :]
+                prediction_interval_result[f"{coverage[i]}_lower"].index = y.index
+                prediction_interval_result[f"{coverage[i]}_upper"].index = y.index
```

Comment:
This is a lot of repeated code, which makes it really hard to read. When we chunk the columns one at a time, any rows that are in `prediction_interval_result` but not in `y` end up being null rows. We also shouldn't have to set the index for each column individually, they're always going to be the same. I understand why we might have to go column by column if there are multiple alphas, but we can split that apart with a few more local variables. I haven't tested it, but I think we can replace this whole chunk with something like:
```
intervals = {
    "lower": result["mean_ci_lower"] - result["mean"],
    "upper": result["mean_ci_upper"] - result["mean"],
}
if len(overlapping_ind) > 0:
    intervals = intervals.iloc[overlapping_ind]
else:
    intervals = intervals[-len(y) : ]
    intervals.index = y.index
prediction_interval_result[f"{coverage[i]}_lower"] = intervals["lower"]
prediction_interval_result[f"{coverage[i]}_upper"] = intervals["upper"]
```